jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml

```diff
@@ -310,6 +310,9 @@
           status-add-test-results: false
           trigger-phrase: 'verify|test'
           cmd: |
+            # verify-godeps compares against upstream, but remote/master might be stale
+            # if .git was retained between runs. Update it explicitly here.
+            git fetch remote master:refs/remotes/remote/master
```

Comment:
No, it fetches the merge branch properly. These are the exact commands run by the Github plugin:

    /usr/bin/git config remote.remote.url https://github.com/kubernetes/kubernetes
    /usr/bin/git -c core.askpass=true fetch --tags --progress https://github.com/kubernetes/kubernetes +refs/heads/*:refs/remotes/upstream/*
    /usr/bin/git config remote.remote1.url https://github.com/kubernetes/kubernetes
    /usr/bin/git -c core.askpass=true fetch --tags --progress https://github.com/kubernetes/kubernetes +refs/pull/31911/merge:refs/remotes/origin/pr/31911/merge

The git plugin doesn't fetch master directly, since it's not obvious that it's necessary -- it thinks it's just testing a specific pull request. Comparing against `remote/master` violates that assumption, obviously.